### PR TITLE
GEODE-4338: Add C++ Continuous Query Example

### PR DESCRIPTION
**Pull request by pdxcodemonkey**

### CODE REVIEW

File: cppcache/include/geode/CqListener.hpp [hunk 1]
```diff
@@ -40,7 +40,13 @@ class CacheListener;
 class APACHE_GEODE_EXPORT CqListener {
  public:
   virtual ~CqListener() noexcept = default;
+
+  /**
+   * @brief constructors
```
Comment (pivotal-jbarrett) [inline cppcache/include/geode/CqListener.hpp]: Obvious comments should be omitted.

File: examples/cpp/CMakeLists.txt [hunk 2]
```diff
@@ -47,6 +47,9 @@ function(add_example)
     DESTINATION examples/cpp/${ADD_EXAMPLE_NAME})
 endfunction()
 
+add_example(NAME continuousquery
```
Comment (pivotal-jbarrett) [inline examples/cpp/CMakeLists.txt]: Formatting seems inconsistent

File: examples/cpp/continuousquery/Order.cpp [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+#include "Order.hpp"
+
+#include <geode/PdxReader.hpp>
+#include <geode/PdxWriter.hpp>
+
+namespace continuousquery {
+
+void Order::fromData(PdxReader& pdxReader) {
+  order_id_ = static_cast<uint32_t>(pdxReader.readLong(ORDER_ID_KEY_));
```
Comment (pivotal-jbarrett) [inline examples/cpp/continuousquery/Order.cpp]: Why muddy the example with types we have to cast?

File: examples/cpp/continuousquery/main.cpp [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+      .setSubscriptionEnabled(true)
+      .create("pool");
+
+  auto regionFactory = cache.createRegionFactory(RegionShortcut::PROXY);
+
+  auto region = regionFactory.setPoolName("pool").create("custom_orders");
+
+  cache.getTypeRegistry().registerPdxType(Order::createDeserializable);
+
+  std::shared_ptr<QueryService> queryService = pool->getQueryService();
```
Comment (pivotal-jbarrett) [inline examples/cpp/continuousquery/main.cpp]: Auto

File: examples/cpp/continuousquery/main.cpp [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+
+  CqAttributesFactory cqFactory;
+
+  std::shared_ptr<MyCqListener> cqListener = std::make_shared<MyCqListener>();
```
Comment (pivotal-jbarrett) [inline examples/cpp/continuousquery/main.cpp]: Auto

File: examples/cpp/continuousquery/main.cpp [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+
+  cqFactory.addCqListener(cqListener);
+  std::shared_ptr<CqAttributes> cqAttributes = cqFactory.create();
```
Comment (pivotal-jbarrett) [inline examples/cpp/continuousquery/main.cpp]: Auto

File: examples/cpp/CMakeLists.txt (hunk 2 above)
Comment (pdxcodemonkey) [reply examples/cpp/CMakeLists.txt]: Removed tabs and fixed indentation

File: examples/cpp/continuousquery/Order.cpp (hunk 3 above)
Comment (pdxcodemonkey) [reply examples/cpp/continuousquery/Order.cpp]: Yeah, kinda silly.  Fixed types, and fixed in other copies of Order.cpp/Order.hpp in examples.

File: examples/cpp/continuousquery/main.cpp (hunk 4 above)
Comment (pdxcodemonkey) [reply examples/cpp/continuousquery/main.cpp]: Fixed

File: examples/cpp/continuousquery/main.cpp (hunk 6 above)
Comment (pdxcodemonkey) [reply examples/cpp/continuousquery/main.cpp]: Fixed

File: examples/cpp/continuousquery/main.cpp [hunk 7]
```diff
@@ -32,65 +33,75 @@ using namespace apache::geode::client;
 using namespace continuousquery;
 
 class MyCqListener : public CqListener {
- public:
-  void onEvent(const CqEvent& cqEvent) {
+public:
+  void onEvent(const CqEvent &cqEvent) override {
```
Comment (pivotal-jbarrett) [inline examples/cpp/continuousquery/main.cpp]: Formatting? Why did you move from right to left spacing? I believe all the other sources and most of Geode is right spaced.

Comment (pdxcodemonkey) [reply examples/cpp/continuousquery/main.cpp]: I ran the clang-format tool in CLion, apparently it didn't do everything "our" way, go figure.

Comment (pivotal-jbarrett) [reply examples/cpp/continuousquery/main.cpp]: If the file has a majority in one direction the formatting rule puts them all to the majority direction. When starting a new file you should prefer right. I suggest updating your formatting rules in CLion to auto format right and then clang format will keep the file in line.
